libraries/AP_Scripting/applets/param-set.lua

```diff
@@ -50,6 +50,71 @@ mavlink:init(5, 1)
 local PARAM_SET_ID = 23
 mavlink:register_rx_msgid(PARAM_SET_ID)
 
+-- support for sending mavlink messages:
+
+local MAV_PARAM_ERROR = {
+    DOES_NOT_EXIST       = 1,
+    VALUE_OUT_OF_RANGE   = 2,
+    PERMISSION_DENIED    = 3,
+    COMPONENT_NOT_FOUND  = 4,
+    READ_ONLY            = 5
+}
+
+-- mavlink message definition
+local param_error_msgid = 345
+local messages = {}
+messages[param_error_msgid] = { -- PARAM_ERROR
+   { "param_index", "<h" },
+   { "target_system", "<B" },
+   { "target_component", "<B" },
+   { "param_id", "<c16" },
+   { "error", "<B" },
+}
+
+function encode(msgid, message, messages_array)
+  local message_map = messages_array[msgid]
+  if not message_map then
+    -- we don't know how to encode this message, bail on it
+    error("Unknown MAVLink message " .. msgid)
+  end
+
+  local packString = "<"
+  local packedTable = {}
+  local packedIndex = 1
+  for i,v in ipairs(message_map) do
+    if v[3] then
+      packString = (packString .. string.rep(string.sub(v[2], 2), v[3]))
+      for j = 1, v[3] do
+        packedTable[packedIndex] = message[message_map[i][1]][j]
+        packedIndex = packedIndex + 1
+      end
+    else
+      packString = (packString .. string.sub(v[2], 2))
+      packedTable[packedIndex] = message[message_map[i][1]]
+      packedIndex = packedIndex + 1
+    end
+  end
+
+  return string.pack(packString, table.unpack(packedTable))
+end
+
+-- send PARAM_ERROR message to GCS
+function send_param_error_response(chan, target_system, target_component, param_id, param_error)
+  -- prepare message
+  local msg = {
+      target_system = target_system,
+      target_component = target_component,
+      param_id = param_id,
+      param_index = -1,
+      error = param_error
+  }
+
+  -- send PARAM_ERROR mavlink message
+  local encoded_msg = encode(param_error_msgid, msg, messages)
+  mavlink.send_chan(chan, param_error_msgid, encoded_msg)
```

Comment:
```suggestion
  mavlink:send_chan(chan, param_error_msgid, encoded_msg)
```
To pass the scripting check.

Reply:
Changed, thanks!